src/shared/logger/winstonToolkitLogger.ts

```diff
@@ -107,12 +107,14 @@ export class WinstonToolkitLogger implements Logger, vscode.Disposable {
         return this.writeToLogs('error', message, ...meta)
     }
 
-    public dispose() {
-        if (!this.disposed) {
-            this.logger.close()
-            this.logger.clear()
-            this.disposed = true
-        }
+    public dispose(): Promise<void> {
+        return this.disposed
+            ? Promise.resolve()
+            : new Promise<void>(resolve => {
+                  this.disposed = true
+                  this.logger.once('finish', resolve)
```

Comment:
Can't find any docs on this "finish" event. This is a non-obvious line of code, worth adding a 1-line comment explaining the intent + a reference.

```suggestion
                  // Trigger winston "finish" event manually. https://example.com/foo/bar #2397
                  this.logger.once('finish', resolve)
```

Reply:
The event is from Node's Stream API which is why winston docs doesn't explicitly mention it. Basically winston (even the logger) is just a wrapper around streams, so many of the same ideas apply.

Oh also `once` is registering an event listener, not triggering anything. Just a Node pattern.